packages/web-forms/src/components/controls/RankControl.vue

```diff
@@ -43,14 +43,20 @@ const values = computed<string[]>({
 			return currentValues.slice();
 		}
 
-		return props.question.currentState.valueOptions.map((option) => option.value);
+		return getRankItems();
 	},
 	set: (orderedValues) => {
 		touched.value = true;
 		props.question.setValues(orderedValues);
 	},
 });
 
+const getRankItems = () => props.question.currentState.valueOptions.map((option) => option.value);
```

Comment:
The `props.question.currentState.valueOptions` contains the original order of the rank's items. The order doesn't get modified. 

The ordered rank's items are assigned to `value` (`props.question.currentState.value`).

When the user taps on the `Rank items` button from the overlay, it takes the rank's items from `props.question.currentState.valueOptions` (original order) and sets it to `value`